File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    
    
    void reverse(vector<int>&nums,int start,int end)
    {
        while(start<end)
        {
            int temp=nums[start];
            nums[start]=nums[end];
            nums[end]=temp;
            start++;
            end--;
            
            
        }
        
    }
    
    
    void rotate(vector<int>& nums, int k) {
        
      k=  k%nums.size();
        
        reverse(nums,0,nums.size()-1);
        
       // cout<<k;  
        reverse(nums,0,k-1);
        // cout<<k;
        
          
        reverse(nums,k,nums.size()-1);
        
        
            
     
       
            
       
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    
    static void reverse(int []nums,int s,int e)
    {
        int start=s;
        int end=e;
        while(start<end)
        {
            int temp=nums[end];
            nums[end]=nums[start];
            nums[start]=temp;
            start++;
            end--;
        }
        
    }
    public void rotate(int[] nums, int k) {
        
        k=k%nums.length;
        int n=nums.length;
        
        reverse(nums,0,n-1);
        reverse(nums,0,k-1);
        reverse(nums,k,n-1);
    }
}